### ISSUE TITLE
Orthofinder stalling at "Analysing Orthogroups: Calculating gene distances" stage

### ISSUE THREAD

**Issue by philoel** [davidemms/OrthoFinder#510]
Hi there! I'm having some sad luck with a big Orthofinder run. This is my first GitHub issue, so I apologize in advance if I share unhelpful info and miss all the actual helpful stuff. I'm happy to say that I have successfully run the example dataset analysis, and I've also used a subset of my current data (about 16 proteomes), and that ran to completion too. Therefore I know all the software is working, and that my proteomes aren't breaking the system. 

A collaborator convinced me to add quite a lot more proteomes to be comparable to his own work, so now I have 55 proteomes, and I'm finding Orthofinder stalls out quite late in the flow, which is confusing. As far as I can tell, all the hard parts are done! I've attached the printed command line output to help suggest what I might do next. 

To summarize my first step, after the run first seemed to have stalled out (my computer went from busy fan sounds to absolute silence for 20hr before I interrupted with option-C), I tried orthofinder -fg (dir) and it seems to be stalling at the same stage (calculating gene distances). 

Thanks for any ideas!
Phil

[Orthofinder_troubleshooting.txt](https://github.com/davidemms/OrthoFinder/files/5980520/Orthofinder_troubleshooting.txt)

**Comment by davidemms** [comment]
Hi Phil

I think the most likely thing is that OrthoFinder was still running the largest tree but that all the others had finished. This would mean that it would have reduced to using one core (OrthoFinder parallelises over the trees but in some cases the largest, which is started first, can still be left running when the remainder have finished). That would explain the fan going quiet. The best thing to do is to use the `top` command to see what's still running, my guess is that OrthoFinder was still waiting for a single instance of FastME. Unfortunately, FastME can sometime take an unexpectedly long amount of time for a tree. With the latests improvements in 2.5.2 using the "-M msa" option is often quicker, while also having the advantage of using MSA tree inference.

You can restart your previous run from the completed orthogroups using "-ft RESULTS_DIR", where RESULTS_DIR is the directory that was being written to when it was interrupted. Let me know if you have any further issues.

All the best
David

**Comment by philoel** [comment]
Hi David,

Thanks for the tips and explanation. 

I tried the top command, and didn't notice anything running by the name of orthofinder, fastME, or anything that sounded involved in these tasks. I am throwing a screen shot up in case you care to inspect. 

Since you suggest that it may be quietly still chugging along on a single tree, I'll let it sit for a day or two more, and then if that doesn't finish things up then I'll try your suggestions, or simply remove some of the extraneous proteomes. I admit I didn't know how to parse the output of the out command, but from what Google just taught me, my computer is indeed not working very hard at the moment. 

Thanks again for the quick response and practical suggestions!
Best,
Phil

![Screen Shot 2021-02-16 at 9 08 31 AM](https://user-images.githubusercontent.com/46672796/108034962-92e5a300-7036-11eb-9b9e-8d29743d46e6.png)

**Comment by davidemms** [comment]
Hi Phil

Looking at your 'top' command it may be a lack of RAM, although I'm not certain about this. It suggests 13G is being used without OrthoFinder using any, with only 3GB left, that's probably too little. This is perhaps the most plausible cause as the OOM Killer might be killing some of OrthoFinder's processes without it knowing.

You could have a look in the directory `OrthoFinder/Results_XXX/WorkingDirectory/Trees_ids/` and see which (if any) of the trees were successfully inferred and which haven't been. I.e. how many tree files are there in the directory, what's the lowest numbered OG with a non-empty tree file and what's the highest numbered OG in there.

I'll have a look to see if there's any way I can detect that threads have been killed in case that's the problem and will submit an enhancement to at least report back to the user if this has occurred. Unfortunately that wouldn't resolve the issue, it that's the problem then you'd need a computer with more RAM. Alternatively, you can reduce the number of species by commenting them out in the WorkingDirectory/SpeciesIDs.txt file and then restarting from the BLAST/DIAMOND results using the '-b' option: `orthofinder -b primary_transcripts/OrthoFinder/Results_Feb13_1`

All the best
David

**Comment by philoel** [comment]
Hi David,

Thanks for the extra tips, this gives me some clear next directions. Since my poor computer might be overwhelmed here, I'll follow your suggestion to comment out a few species. The flow worked very nicely for a subset of these species already, so in the end if the problem is with my hardware then I guess there is nothing left to explore as an issue here on Github. 

Thanks for the great support (and lovely + well-documented package)! I'm not typically good with phylogenetic softwares or any form of programming, yet I feel pretty good working with Orthofinder :)

Have a great weekend,
Phil

**Comment by davidemms** [comment]
Thanks Phil, that's great to hear!
